pkg/distributor/distributor.go

```diff
@@ -241,196 +241,206 @@ func (d *Distributor) stopping(_ error) error {
 
 func (d *Distributor) Push(ctx context.Context, grpcReq *connect.Request[pushv1.PushRequest]) (*connect.Response[pushv1.PushResponse], error) {
 	req := &distributormodel.PushRequest{
-		Series: make([]*distributormodel.ProfileSeries, 0, len(grpcReq.Msg.Series)),
+		Series:         make([]*distributormodel.ProfileSeries, 0, len(grpcReq.Msg.Series)),
+		RawProfileType: distributormodel.RawProfileTypePPROF,
 	}
 
 	for _, grpcSeries := range grpcReq.Msg.Series {
-		series := &distributormodel.ProfileSeries{
-			Labels:  grpcSeries.Labels,
-			Samples: make([]*distributormodel.ProfileSample, 0, len(grpcSeries.Samples)),
-		}
 		for _, grpcSample := range grpcSeries.Samples {
 			profile, err := pprof.RawFromBytes(grpcSample.RawProfile)
 			if err != nil {
 				return nil, connect.NewError(connect.CodeInvalidArgument, err)
```

Comment:
I wonder if this should keep the error around and continue the loop and return the later after all other samples are pushd

Reply:
https://github.com/grafana/pyroscope/pull/4342/commits/d615aef13117389ca18fd20e54c02d3aa7941da5